google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -51,11 +56,108 @@ public class StreamWriter implements AutoCloseable {
    */
   private final String writerId = UUID.randomUUID().toString();
 
+  /**
+   * Stream can access a single connection or a pool of connection depending on whether multiplexing
+   * is enabled.
+   */
+  private final SingleConnectionOrConnectionPool singleConnectionOrConnectionPool;
+
+  /**
+   * Static map from {@link ConnectionPoolKey} to connection pool. Note this map is static to be
+   * shared by every stream writer in the same process.
+   */
+  private static final Map<ConnectionPoolKey, ConnectionWorkerPool> connectionPoolMap =
```

Comment:
I thought we talked about moving this pool map onto ConnectionWorkerPool?

Reply:
I think putting here will make the structure a lot easier to understand, otherwise we need to create a separate class e.g. PoolCollection to store the mapping from poolkey to connection pool